lib/ansible/modules/cloud/amazon/api_gateway.py

```diff
@@ -0,0 +1,232 @@
+                    'version': '0.1'}
+
+DOCUMENTATION = '''
+---
+module: api-gateway
+short_description: Manage AWS API Gateway APIs
+description:
+     - Allows for the management of API Gatway APIs
+     - Normally you should give the api_id since there is no other
+       stable guaranteed unique identifier for the API.  If you do
+       not give api_id then a new API will be create each time
+       this is run.
+     - Beware that there are very hard limits on the rate that
+       you can call API Gateway's REST API.  You may need to patch
+       your boto.  See https://github.com/boto/boto3/issues/876
+       and discuss with your AWS rep.
+version_added: '2.3'
+requirements: [ boto3 ]
+options:
+  api_id:
+    description:
+      - The ID of the API you want to manage
+    required: false
+  state:
+    description:
+      - NOT IMPLEMENTED Create or delete API - currently we always create
+    required: false
+    default: present
+    choices: [ 'present', 'absent' ]
+  swagger_file:
+    description:
+      - JSON or YAML file containing swagger definitions for API
+    required: false
+  swagger:
+    description:
+      - Swagger definitions for API in JSON or YAML as a string direct from playbook
+    required: false
+  stage:
+    description:
+      - stage API should be deployed to
+    required: false
+author:
+    - 'Michael De La Rue (@mikedlr)'
+extends_documentation_fragment:
+    - aws
+    - ec2
+notes:
+   - a future version of this module will probably use tags or another
+     ID so that an API can be create only once.
+   - as an early work around an intermediate version will probably do
+     the same using a tag embedded in the API name.
+
+'''
+
+EXAMPLES = '''
+# Update API resources for development
+tasks:
+- name: update API
+  api_gateway:
+    api_id: 'abc123321cba'
+    state: present
+    swagger_file: my_api.yml
+
+# update definitions and deploy API to production
+tasks:
+- name: deploy API
+  api_gateway:
+    api_id: 'abc123321cba'
+    state: present
+    swagger_file: my_api.yml
+    stage: production
+    deploy_desc: Make auth fix available.
+'''
+
+RETURN = '''
+output:
+  description: the data returned by put_restapi in boto3
+  returned: success
+  type: dict
+  sample:
+    'data':
+      {
+          "id": "abc123321cba",
+          "name": "MY REST API",
+          "createdDate": 1484233401
+      }
+'''
+
+import os
+
+try:
+    import botocore
+    HAS_BOTOCORE = True
+except ImportError:
+    HAS_BOTOCORE = False
+
+try:
+    import boto3
+    HAS_BOTO3 = True
+except ImportError:
+    HAS_BOTO3 = False
+
+
+def main():
+    argument_spec = ec2_argument_spec()
+    argument_spec.update(
+        dict(
+            api_id=dict(type='str', required=False),
+            state=dict(type='str', default='present', choices=['present', 'absent']),
+            swagger_file=dict(type='str', default=None, aliases=['src', 'api_file']),
+            swagger=dict(type='str', default=None),
```

Comment:
This type can actually be `json` which will accept any of:

```
swagger: |
  { .... contents
  ... in multiline strings }
swagger: > 
  {... contents
  .. in multiline, but joined to a single one }
swagger:
  value: in yaml format and parsed to a dict
swagger: {json: style object}
```

Which makes it easy for users to provide the swagger doc if they want to use strings from some source or if they want to include variables in Ansible. Once you switch that, you can change line 188 to be `apidata = json.dumps(swagger)` because Ansible will decode all 4 of the above formats into a standard Python dictionary.